### ISSUE TITLE
Incorporate polar onion peeling algorithm into PyAbel

### ISSUE THREAD

**Issue by DanHickstein** [PyAbel/PyAbel#30]
It would be nice to implement one of the inverse Abel transform methods that uses a polar basis set to allow for smooth transforms of noisy data. [Polar onion peeling (POP)](http://dx.doi.org/10.1063/1.3126527) is one of the most popular methods for this. 

There is a nice Matlab implementation of the POP algorithm [here](http://www.mathworks.com/matlabcentral/fileexchange/41064-polar-onion-peeling). Adi Natan, the author of that code, said that it would be fine if we ported it to python and incorporate it in to PyAbel. He also has the matlab code that generates the basis sets, so we would not be limited to the 1024x1024 basis set that comes with the program.

**Comment by rth** [comment]
That would be great! It would be interesting  to see how different implementations behave with noisy data.

**Comment by DanHickstein** [comment]
Now that #270 has added rBasex, there is probably little motivation to also add pBasex.

**Comment by MikhailRyazanov** [comment]
Now I need to say here that POP is not really correct. Namely, this procedure:
![image](https://user-images.githubusercontent.com/12854133/79825006-07877c80-8355-11ea-92ac-b3e819449238.png)
that generates the projection of an anisotropic ring is valid only if the ring is infinitely narrow (that is, _b_<sub>_r_</sub>(_R_) → Abel[δ(_R_ − _r_)]). In reality, for extended basis functions with different angular dependences, their projections have different radial parts, not just by the (_R_/_r_)<sup>_n_</sup> factor.

Here is a comparison of the actual projections and their POP approximations for the triangular basis functions used in rBasex (the POP article is not clear about the shape of their basis functions, but they are unlikely to be narrower than this):
![POP](https://user-images.githubusercontent.com/12854133/79825885-7d8ce300-8357-11ea-820c-1a1e02ec3837.png)
_p_<sub>6; 0</sub>(_R_) here is what they call _b_<sub>6</sub>(_R_) — projection of the isotropic basis function centered at _r_ = 6, and _p_<sub>6; _n_</sub>(_R_) are projections of _anisotropic_ basis functions with cos<sup>_n_</sup> θ angular dependences.
As can be seen, all POP approximations have noticeable errors, larger for higher angular orders.

However, for larger radii _r_, the relative widths of the basis functions decrease, so they become effectively narrower, and the POP approximation works better. Perhaps, for real images (with min _r_ ≳ 100 px), the results will be reasonable enough (plus, the authors say that their method has some other problems at low radii).

One potential advantage is that they use Legendre polynomials instead of cosine powers, so the method should be more numerically stable for high angular orders. Although I don't know whether this is really true and whether this higher stability overweighs the increasing approximation errors. If somebody wants to compare, it would be interesting to see the results...

**Comment by DanHickstein** [comment]
Okay, this is interesting stuff. Is your conclusion is the same as mine: incorporating POP might be interesting, but it's a fairly low priority to incorporate it?

**Comment by MikhailRyazanov** [comment]
Yes, maybe label this issue as low-priority, but I wouldn't close it.

**Comment by adinatan** [comment]
Just to add that unlike the paper mentioned,  the error that is accumulated closer to the origin at small radii  is dealt in the matlab code by restricting the leading beta order that can be used, which created an effective filter for the angle resolution, as well as attenuate the signal the closer it gets to the origin. In addition, this behavior is negligible for radius ~ >100 pixels, so it can be easily circumvented  by resampling the domain needed such that the error is restricted to an unimportant neighborhood around the origin.

**Comment by MikhailRyazanov** [comment]
I guess, if this is ever implemented, it would be nice to have the reference implementation and add any possible improvements as options (maybe enabled by default). In any case, examining results from existing implementations beforehand should help in judging what to do and how. Maybe these deviations are negligible in practice.
